easypermissions/src/main/java/pub/devrel/easypermissions/AppSettingsDialog.java

```diff
@@ -121,45 +127,121 @@ public Builder(@NonNull Activity activity, @NonNull String rationale) {
 
         /**
          * Create a new Builder for an {@link AppSettingsDialog}.
-         * @param fragment the Fragment in which to display the dialog.
+         *
+         * @param fragment  the Fragment in which to display the dialog.
          * @param rationale text explaining why the user should launch the app settings screen.
+         * @deprecated Use {@link #Builder(android.support.v4.app.Fragment)} with {@link
+         * #setRationale(String)} or {@link #setRationale(int)}.
```

Comment:
The reason to include rationale in the builder constructor is that it implies that rationale is the only **required** option, the rest of the setters are optional.

I see that you're adding a default message.  That should be OK.